lib/cisco_node_utils/vxlan_global.rb

```diff
@@ -0,0 +1,158 @@
+      host_moves
+    end
+
+    def dup_host_ip_addr_detection_timeout
+      _host_moves, timeout = dup_host_ip_addr_detection
+      return default_dup_host_ip_addr_detection_timeout if timeout.nil?
+      timeout
+    end
+
+    def dup_host_ip_addr_detection_set(state, host_moves, timeout)
+      set_args = {}
+      set_args[:state] = state
+      set_args[:host_moves] = host_moves
+      set_args[:timeout] = timeout
+
+      config_set('vxlan_global', 'dup_host_ip_addr_detection', set_args)
+    end
+
+    def default_dup_host_ip_addr_detection
+      [default_dup_host_ip_addr_detection_host_moves,
+       default_dup_host_ip_addr_detection_timeout]
+    end
+
+    def default_dup_host_ip_addr_detection_host_moves
+      config_get_default('vxlan_global',
+                         'dup_host_ip_addr_detection_host_moves')
+    end
+
+    def default_dup_host_ip_addr_detection_timeout
+      config_get_default('vxlan_global', 'dup_host_ip_addr_detection_timeout')
+    end
+
+    # dup-host-mac-detection
+    def dup_host_mac_detection
+      get_args = {}
+      match = config_get('vxlan_global', 'dup_host_mac_detection', get_args)
+      if match.nil? || match.first.nil?
+        default_dup_host_mac_detection
+      else
+        match.first.collect(&:to_i)
+      end
+    end
+
+    def dup_host_mac_detection_host_moves
+      host_moves, _timeout = dup_host_mac_detection
+      return default_dup_host_mac_detection_host_moves if host_moves.nil?
+      host_moves
+    end
+
+    def dup_host_mac_detection_timeout
+      _host_moves, timeout = dup_host_mac_detection
+      return default_dup_host_mac_detection_timeout if timeout.nil?
+      timeout
+    end
+
+    def dup_host_mac_detection_set(host_moves, timeout)
+      set_args = {}
+      set_args[:host_moves] = host_moves
+      set_args[:timeout] = timeout
+
+      config_set('vxlan_global', 'dup_host_mac_detection', set_args)
+    end
+
+    def dup_host_mac_detection_default
+      config_set('vxlan_global', 'dup_host_mac_detection_default')
+    end
+
+    def default_dup_host_mac_detection
+      [default_dup_host_mac_detection_host_moves,
+       default_dup_host_mac_detection_timeout]
+    end
+
+    def default_dup_host_mac_detection_host_moves
+      config_get_default('vxlan_global', 'dup_host_mac_detection_host_moves')
+    end
+
+    def default_dup_host_mac_detection_timeout
+      config_get_default('vxlan_global', 'dup_host_mac_detection_timeout')
+    end
+
+    # anycast-gateway-mac
+    def anycast_gateway_mac
+      mac_addr = config_get('vxlan_global', 'anycast_gateway_mac')
+      mac_addr.nil? ? ' ' : mac_addr.first
+    end
+
+    def anycast_gateway_mac_set(state, mac_addr)
+      fail TypeError unless mac_addr.is_a?(String)
+      if state == 'no'
+        config_set('vxlan_global', 'anycast_gateway_mac', 'no', ' ')
+      else
+        config_set('vxlan_global', 'anycast_gateway_mac', ' ', mac_addr)
```

Comment:
Use the key-value style (per earlier comment).

Also a nit: use empty strings `''` instead of `' '`